docs/build_pipeline_controller.md

```diff
@@ -0,0 +1,37 @@
+<div align="center"><h1>BuildPipeline Controller</h1></div>
+
+```mermaid
+
+%%{init: {'theme':'forest'}}%%
+flowchart TD
+%% Defining the styles
+  classDef Red fill:#FF9999;
+  classDef Amber fill:#FFDEAD;
+  classDef Green fill:#BDFFA4;
+ 
+  %% Node definitions
+predicate((PREDICATE: <br> filter events related to <br> PipelineRun <br> that are signed and <br> have succeeded))
+get_pipeline_run{Pipeline found?}
+retrieve_associated_entity(Retrieve the entity <br> component/application)
+determine_snapshot{does a snapshot exist?}
+create_snapshot(gather components <br> create snapshot <br> add snapshot)
```

Comment:
nit: probably better to start the sentence with an uppercase letter.

Reply:
Also, the `add snapshot` part is probably not needed, maybe something like this:
```
Gather Application Components<br>Add new Component image<br>Create Snapshot
```